app.py

```
@@ -546,27 +550,42 @@ def upload():
-    valid_files = [file for file in all_files if file and file.filename]
-
-    if not valid_files:
-        flash("No valid files or folder selected", "error")
-        return redirect(url_for("upload"))
-
-    # Create user-specific folder
-    user_folder = create_user_folder(current_user.username)
-
-    for file in valid_files:
+# Add rate limiting for Gemini API
+from threading import Lock
```

Ruff (0.8.2)

Redefinition of unused `Lock` from line 70

Remove definition: `Lock`

(F811)